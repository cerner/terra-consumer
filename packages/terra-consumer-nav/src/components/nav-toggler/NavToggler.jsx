import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import Toggler from 'terra-toggle';
import Arrange from 'terra-arrange';
import IconChevronDown from 'terra-icon/lib/icon/IconChevronDown';
import IconChevronUp from 'terra-icon/lib/icon/IconChevronUp';
import styles from './NavToggler.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Element to be displayed on top of the button that opens the toggle content.
   */
  header: PropTypes.element,

  /**
   * Function callback when the toggle component is clicked.
   */
  handleToggle: PropTypes.func.isRequired,

  /**
   * Whether or not the content should be toggled open or not.
   */
  isOpen: PropTypes.bool,

  /**
   * Items to be displayed within the toggler.
   */
  children: PropTypes.node,
};

const defaultProps = {
  isOpen: false,
};

const NavToggler = ({
  header,
  handleToggle,
  isOpen,
  children,
  ...customProps
}) => {
  const toggleIcon = isOpen ? <IconChevronUp /> : <IconChevronDown />;

  return (
    <div {...customProps}>
      <button className={cx('toggle-header')} onClick={() => handleToggle(!isOpen)}>
        <Arrange
          alignFill="stretch"
          alignFitEnd="center"
          fill={header}
          fitEnd={toggleIcon}
        />
      </button>
      <Toggler isOpen={isOpen} isAnimated className={cx('toggler')}>
        {children}
      </Toggler>
    </div>
  );
};

NavToggler.propTypes = propTypes;
NavToggler.defaultProps = defaultProps;

export default NavToggler;
