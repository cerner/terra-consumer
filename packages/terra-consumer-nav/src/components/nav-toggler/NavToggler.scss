@import '../../variables';

:local {
  .toggle-header {
    background-color: transparent;
    border: 0;
    color: inherit;
    cursor: pointer;
    display: block;
    font-size: inherit;
    line-height: inherit;
    padding: 0;
    padding-right: 20px;
    text-align: left;
    text-decoration: none;
    width: 100%;

    &:hover {
      background: var(--terra-consumer-toggler-hover-background, rgba(255, 255, 255, 0.4));
      color: var(--terra-consumer-nav-item-color);
      @media screen and (max-width: $terra-medium-breakpoint) {
        color: #fff;
      }
    }
  }

  .toggler {
    padding: 0;
  }
}
