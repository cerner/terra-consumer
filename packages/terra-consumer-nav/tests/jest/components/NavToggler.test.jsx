import React from 'react';
import NavToggler from '../../../src/components/nav-toggler/NavToggler';

const testData = {
  header: <span>Test</span>,
  handleToggle: () => {},
  isOpen: false,
};

describe('Nav Toggler', () => {
  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(
      <NavToggler {...testData}>
        <div>
          <p>Test Sub element 1</p>
          <p>Test Sub element 2</p>
        </div>
      </NavToggler>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should apply custom classes', () => {
    const wrapper = shallow(
      <NavToggler {...testData} className="test-class" >
        <div>test</div>
      </NavToggler>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
