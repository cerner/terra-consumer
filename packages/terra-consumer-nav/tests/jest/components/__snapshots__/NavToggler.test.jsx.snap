// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Toggler should apply custom classes 1`] = `
<div
  className="test-class"
>
  <button
    className="toggle-header"
    onClick={[Function]}
  >
    <Arrange
      alignFill="stretch"
      alignFitEnd="center"
      fill={
        <span>
          Test
        </span>
      }
      fitEnd={
        <IconChevronDown
          className=""
          data-name="Layer 1"
          isBidi={true}
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        />
      }
    />
  </button>
  <Toggle
    className="toggler"
    isAnimated={true}
    isOpen={false}
  >
    <div>
      test
    </div>
  </Toggle>
</div>
`;

exports[`Nav Toggler should render a default component 1`] = `
<div>
  <button
    className="toggle-header"
    onClick={[Function]}
  >
    <Arrange
      alignFill="stretch"
      alignFitEnd="center"
      fill={
        <span>
          Test
        </span>
      }
      fitEnd={
        <IconChevronDown
          className=""
          data-name="Layer 1"
          isBidi={true}
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        />
      }
    />
  </button>
  <Toggle
    className="toggler"
    isAnimated={true}
    isOpen={false}
  >
    <div>
      <p>
        Test Sub element 1
      </p>
      <p>
        Test Sub element 2
      </p>
    </div>
  </Toggle>
</div>
`;
